test/correctness/simd_op_check.cpp

```diff
@@ -1685,6 +1685,11 @@ void check_hvx_all() {
     check("vabs(v*.h)", hvx_width/2, abs(i16_1));
     check("vabs(v*.w)", hvx_width/4, abs(i32_1));
 
+    check("vmpa(v*.ub,r*.b)", hvx_width/1, i16(u8_1)*2 + i16(u8_2)*3);
+    check("vmpa(v*.ub,r*.b)", hvx_width/1, i16(u8_1)*2 + 3*i16(u8_2));
+    check("vmpa(v*.ub,r*.b)", hvx_width/1, 2*i16(u8_1) + 3*i16(u8_2));
+    check("v*.h += vmpa(v*.ub,r*.b)", hvx_width/1, 2*i16(u8_1) + 3*i16(u8_2) + i16(u8_3));
```

Comment:
sorry, my bad. You are right. I was playing with teh wrong vector factor in my head.